drivers/include/periph/sdmmc.h

```diff
@@ -0,0 +1,1624 @@
+     * It is called by the high-level SDIO/SD/MMC device API function
+     * @ref sdmmc_xfer before it sends a block-oriented, stream or
+     * byte/multi-byte command to the card to start the transfer from or
+     * to the card.
+     *
+     * A typical activity of this function is the configuration of
+     * the DMA transfer.
+     *
+     * @see @ref sdmmc_xfer
+     * @pre @p dev and @p xfer must not be `NULL`.
+     *
+     * @param[in]   dev     SDIO/SD/MMC device to be used
+     * @param[in]   xfer    Transfer descriptor of type @ref sdmmc_xfer_desc_t
+     *
+     * @return  0           on success
+     * @return  -EINVAL     on invalid transfer parameters
+     * @return  -EIO        on not further specified error incl. hardware errors
+     */
+    int (*xfer_prepare)(sdmmc_dev_t *dev, sdmmc_xfer_desc_t *xfer);
+
+    /**
+     * @brief   Execute the data transfer
+     *
+     * Low-level SDIO/SD/MMC peripheral driver function to transfer the data
+     * (read or write) by the SDIO/SD/MMC peripheral.
+     *
+     * It is called by the high-level SDIO/SD/MMC device API function
+     * @ref sdmmc_xfer after it sent the block-oriented, stream or
+     * byte/multi-byte command to start the transfer from or to the card.
+     *
+     * The function returns the number of transferred blocks in @p done.
+     * Most SDIO/SD/MMC peripherals use a block or byte counter when
+     * transferring data, which can be used for this purpose.
+     *
+     * @note If the SDIO/SD/MMC peripheral does not allow to determine
+     *       the number of transferred blocks, the function must return `0`
+     *       in @p done in case of an error or @ref sdmmc_xfer_desc_t::block_num
+     *       on success.
+     *
+     * @see @ref sdmmc_xfer
+     * @pre @p dev and @p xfer must not be `NULL`. \n
+     *      @p data_rd must not be `NULL` for read transfers and \n
+     *      @p data_wr must not be `NULL` for write transfers.
+     *
+     * @param[in]   dev         SDIO/SD/MMC device to be used
+     * @param[in]   xfer        Transfer descriptor of type @ref sdmmc_xfer_desc_t
+     * @param[in]   data_wr     Buffer with data to write in write transfers, NULL otherwise
+     * @param[out]  data_rd     Buffer for data to read in read transfers, NULL otherwise
+     * @param[out]  done        Number of blocks transferred, can be `NULL`
+     *
+     * @return  0           on success
+     * @return  -ENODEV     if card is not present or not usable
+     * @return  -ENOTSUP    if card does not support a used command or is in wrong state
+     * @return  -EBUSY      if card is busy
+     * @return  -EFAULT     on card status error
+     * @return  -ETIMEDOUT  on timeout condition
+     * @return  -EINVAL     on invalid transfer parameters
+     * @return  -EBADMSG    on CRC7 error in data
+     * @return  -ENOMEM     on RX FIFO overflow or TX FIFO underrun error
+     * @return  -EIO        on not further specified error incl. hardware errors
```

Comment:
```suggestion
     * @retval  0           on success
     * @retval  -ENODEV     if card is not present or not usable
     * @retval  -ENOTSUP    if card does not support a used command or is in wrong state
     * @retval  -EBUSY      if card is busy
     * @retval  -EFAULT     on card status error
     * @retval  -ETIMEDOUT  on timeout condition
     * @retval  -EINVAL     on invalid transfer parameters
     * @retval  -EBADMSG    on CRC7 error in data
     * @retval  -ENOMEM     on RX FIFO overflow or TX FIFO underrun error
     * @retval  -EIO        on not further specified error incl. hardware errors
```

Doxygen will generate a fancy table mapping return values to their description when using `@retval`. The rule of thumb is to use `@return` for a general description of the return value, and `@retval` to map specific return values to their meaning.